api/java/org/openyolo/api/internal/CredentialRetrieveActivity.java

```diff
@@ -106,23 +113,27 @@ public void onResponse(long queryId, List<QueryResponse> queryResponses) {
 
                 protoResponses.put(queryResponse.responderPackage, response);
 
-                if (response.getRetrieveIntent().isEmpty()) {
-                    continue;
-                }
 
-                Intent retrieveIntent;
-                try {
-                    retrieveIntent =
-                            IntentUtil.fromBytes(response.getRetrieveIntent().toByteArray());
-                } catch (BadParcelableException ex) {
-                    Log.w(LOG_TAG, "Failed to parse intent from bytes");
+                // a provider indicates that it has credentials via the credentials_available field
+                // on the response proto. Prior 0.3.0, the provider would directly return an intent.
+                // For backwards compatibility this is currently inspected as a fallback for the
+                // absence of the credentials_available field, but the Intent itself is no longer
+                // used.
+                // TODO: remove backwards compatibility with retrieve_intent after 0.3.0
+                if (!response.getCredentialsAvailable()
+                        && !response.getRetrieveIntent().isEmpty()) {
                     continue;
                 }
 
-                if (!queryResponse.responderPackage.equals(
-                        retrieveIntent.getComponent().getPackageName())) {
-                    Log.w(LOG_TAG, "Package mismatch between provider and retrieve intent");
-                    continue;
+                Intent retrieveIntent = new Intent(ProtocolConstants.RETRIEVE_CREDENTIAL_ACTION);
```

Comment:
We are deprecating retrieve intent, but here it will be just ignored. So we are breaking the compatibility if a provider don't have the action `org.openyolo.credential.retrieve`, as it was not require before (and it's my case).

I think we should have both, and prefer the action `org.openyolo.credential.retrieve` if the provider did set `getCredentialsAvailable()`.

Reply:
OK, we'll support the old intent for 0.3.0 but please do add support for direct invocation of the retrieve flow ASAP. It's actually required in the spec:

http://openid.net/specs/openyolo-android-ID1.html#retrieving-credentials

Reply:
Yes you're right, I've made the change on my side. It will be live in ~ 2 weeks.